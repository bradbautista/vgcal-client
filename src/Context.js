import React from 'react'

const Context = React.createContext({
    releases: [],    
})

export default Context