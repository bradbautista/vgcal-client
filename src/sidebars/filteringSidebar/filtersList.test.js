import React from 'react';
import ReactDOM from 'react-dom';
import FiltersList from './filtersList';
import Context from '../../Context';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

it('Renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Context.Provider value={ { checkboxes: [] }}><FiltersList /></Context.Provider>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('Renders the filters list as expected', () => {
    
    const tree = renderer
        .create(
            
            <BrowserRouter>
                <Context.Provider value={ { checkboxes: [] }}>
                    <FiltersList />
                </Context.Provider>
            </BrowserRouter>
            )
        .toJSON();
    expect(tree).toMatchSnapshot();
        
})