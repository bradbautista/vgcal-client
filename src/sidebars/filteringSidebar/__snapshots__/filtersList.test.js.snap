// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the filters list as expected 1`] = `
<div
  className="filtering-container"
>
  <h2
    className="filters-header"
  >
    Filtering options
  </h2>
  <div>
    Note: If you check PC and Switch, you will get games which are coming out on PC 
    <em>
      and
    </em>
     Switch, not PC 
    <em>
      or
    </em>
     Switch.
  </div>
  <label
    className="checkbox-wrapper"
    htmlFor="Xbox"
  >
    <input
      checked={false}
      id="Xbox"
      name="Xbox"
      onChange={[Function]}
      type="checkbox"
      value="Xbox"
    />
    <span
      className="checkmark"
    />
    Xbox
  </label>
  <label
    className="checkbox-wrapper"
    htmlFor="PlayStation"
  >
    <input
      checked={false}
      id="PlayStation"
      name="PlayStation"
      onChange={[Function]}
      type="checkbox"
      value="PlayStation"
    />
    <span
      className="checkmark"
    />
    PlayStation
  </label>
  <label
    className="checkbox-wrapper"
    htmlFor="PC"
  >
    <input
      checked={false}
      id="PC"
      name="PC"
      onChange={[Function]}
      type="checkbox"
      value="PC"
    />
    <span
      className="checkmark"
    />
    PC
  </label>
  <label
    className="checkbox-wrapper"
    htmlFor="Switch"
  >
    <input
      checked={false}
      id="Switch"
      name="Switch"
      onChange={[Function]}
      type="checkbox"
      value="Switch"
    />
    <span
      className="checkmark"
    />
    Switch
  </label>
  <label
    className="checkbox-wrapper"
    htmlFor="Mac"
  >
    <input
      checked={false}
      id="Mac"
      name="Mac"
      onChange={[Function]}
      type="checkbox"
      value="Mac"
    />
    <span
      className="checkmark"
    />
    Mac
  </label>
  <label
    className="checkbox-wrapper"
    htmlFor="Android"
  >
    <input
      checked={false}
      id="Android"
      name="Android"
      onChange={[Function]}
      type="checkbox"
      value="Android"
    />
    <span
      className="checkmark"
    />
    Android
  </label>
  <label
    className="checkbox-wrapper"
    htmlFor="iPhone"
  >
    <input
      checked={false}
      id="iPhone"
      name="iPhone"
      onChange={[Function]}
      type="checkbox"
      value="iPhone"
    />
    <span
      className="checkmark"
    />
    iPhone
  </label>
  <label
    className="checkbox-wrapper"
    htmlFor="iPad"
  >
    <input
      checked={false}
      id="iPad"
      name="iPad"
      onChange={[Function]}
      type="checkbox"
      value="iPad"
    />
    <span
      className="checkmark"
    />
    iPad
  </label>
</div>
`;
